src/platform/site-wide/mega-menu/containers/Main.jsx

```diff
@@ -38,7 +40,32 @@ export function getAuthorizedLinkData(
   ];
 }
 
-export class Main extends React.Component {
+export class Main extends Component {
+  static propTypes = {
+    megaMenuData: PropTypes.arrayOf(
+      PropTypes.shape({
+        href: PropTypes.string,
+        menuSections: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+      }).isRequired,
+    ).isRequired,
+    toggleMobileDisplayHidden: PropTypes.func.isRequired,
+    togglePanelOpen: PropTypes.func.isRequired,
+    updateCurrentSection: PropTypes.func.isRequired,
+    // From mapStateToProps.
+    currentDropdown: PropTypes.string,
+    currentSection: PropTypes.string,
+    data: PropTypes.arrayOf(
+      PropTypes.shape({
+        href: PropTypes.string,
+        menuSections: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+        title: PropTypes.string.isRequired,
+      }).isRequired,
+    ).isRequired,
+    display: PropTypes.object,
+    loggedIn: PropTypes.bool.isRequired,
+    showDashboard2: PropTypes.bool.isRequired,
+  };
```

Comment:
If these prop types make you feel overwhelmed, good! Me, too! haha There's a lot of refactoring that can and should happen in the future for the mega menu I think 🙂